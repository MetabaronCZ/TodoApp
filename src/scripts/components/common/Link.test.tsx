import React from 'react';

import { render } from '@testing-library/react';
import { describe, expect, it } from '@jest/globals';

import { Link } from 'components/common/Link';
import { withMockedProviders } from 'test/component';

describe('components/common/Link', () => {
  it('should render correctly', () => {
    const tree = render(withMockedProviders(<Link href="/test">CONTENT</Link>));
    expect(tree.container).toMatchSnapshot();
    expect(tree.getByRole('link')).toBeInTheDocument();
    expect(tree.getByText('CONTENT')).toBeInTheDocument();
  });

  it('should render props', () => {
    const { container } = render(
      withMockedProviders(<Link href="/test" target="_blank" />),
    );
    expect(container).toMatchSnapshot();
    expect(container.firstChild).toHaveAttribute('href', '/test');
    expect(container.firstChild).toHaveAttribute('target', '_blank');
  });
});
