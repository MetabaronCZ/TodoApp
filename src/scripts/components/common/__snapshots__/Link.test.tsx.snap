// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components/common/Link > should render correctly 1`] = `
.c0 {
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #22c;
}

.c0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0:focus {
  outline: 1px dotted #22c;
  outline-offset: -1px;
}

<div>
  <a
    class="c0"
    href="/test"
  >
    CONTENT
  </a>
</div>
`;

exports[`components/common/Link > should render props 1`] = `
.c0 {
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #22c;
}

.c0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0:focus {
  outline: 1px dotted #22c;
  outline-offset: -1px;
}

<div>
  <a
    class="c0"
    href="/test"
    target="_blank"
  />
</div>
`;
